diff --git a/common-src/glib-util.c b/common-src/glib-util.c
index 6a42e8e..99f55f7 100644
--- a/common-src/glib-util.c
+++ b/common-src/glib-util.c
@@ -38,12 +38,8 @@ glib_init(void) {
     if (did_glib_init) return;
     did_glib_init = TRUE;
 
-    /* set up libcurl (this must happen before threading 
-     * is initialized) */
+    /* set up libcurl */
 #ifdef HAVE_LIBCURL
-# ifdef G_THREADS_ENABLED
-    g_assert(!g_thread_supported()); /* assert threads aren't initialized yet */
-# endif
     g_assert(curl_global_init(CURL_GLOBAL_ALL) == 0);
 #endif
 
@@ -64,12 +60,6 @@ glib_init(void) {
     /* Initialize glib's type system.  On glib >= 2.24, this will initialize
      * threads, so it must be done after curl is initialized. */
     g_type_init();
-
-    /* And set up glib's threads */
-#if defined(G_THREADS_ENABLED) && !defined(G_THREADS_IMPL_NONE)
-    if (!g_thread_supported())
-	g_thread_init(NULL);
-#endif
 }
 
 typedef enum {
